app/assets/src/components/common/DetailsSidebar/PipelineStepDetailsMode/PipelineStepDetailsMode.jsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import cs from "./pipeline_step_details_mode.scss";
+
+class PipelineStepDetailsMode extends React.Component {
+  renderInputFiles() {
+    const { inputFiles } = this.props;
+    if (inputFiles && inputFiles.length) {
+      const fromStepNameToFile = {};
+      inputFiles.forEach(file => {
+        if (!(file.fromStepName in fromStepNameToFile)) {
+          fromStepNameToFile[file.fromStepName] = [];
+        }
+        fromStepNameToFile[file.fromStepName].push(file);
+      });
+
+      const fileGroupList = Object.keys(fromStepNameToFile).map(
+        fromStepName => {
+          const fileGroup = fromStepNameToFile[fromStepName];
+          const fileList = fileGroup.map((file, i) => {
+            return (
+              <div className={cs.link} key={file.fileName + i.toString()}>
+                {file.fileName}
+              </div>
+            );
+          });
+
+          // TODO(ezhong): Figure out what to put as fileGroupHeader if input is
+          // provided by a user (instead of step output)
+          return (
+            <div className={cs.inputFileGroup} key={fromStepName}>
+              <div
+                className={cs.fileGroupHeader}
+              >{`From ${fromStepName}:`}</div>
+              {fileList}
+            </div>
+          );
+        }
+      );
+      return (
+        <div className={cs.block}>
```

Comment:
Might want to use more explicit names. Make it easy to track between css and js.
E.g. `stepfilesListBox`
Same for all other names...